src/gt4py/stencil_object.py

```diff
@@ -132,11 +137,13 @@ def _get_max_domain(self, field_args, origin):
         -------
             `Shape`: the maximum domain size.
         """
-        max_domain = Shape([np.iinfo(np.uintc).max] * self.domain_info.ndims)
-        shapes = {name: Shape(field.shape) for name, field in field_args.items()}
-        for name, shape in shapes.items():
-            upper_boundary = Index(self.field_info[name].boundary.upper_indices)
-            max_domain &= shape - (Index(origin[name]) + upper_boundary)
+        large_val = np.iinfo(np.uintc).max
+        max_domain = Shape([large_val] * self.domain_info.ndims)
+        for name, field in field_args.items():
+            field_mask = self._get_field_mask(field)
+            upper_boundary = self.field_info[name].boundary.upper_indices.filter_mask(field_mask)
+            field_domain = Shape(field.shape) - (origin[name] + upper_boundary)
+            max_domain &= Shape.from_mask(field_domain, field_mask, default=large_val)
         return max_domain
 
     def _validate_args(self, used_field_args, used_param_args, domain, origin):
```

Comment:
It is never checked if the provided args actually have the right mask to go with the axes used in the stencil definition.

Reply:
Good catch. This is included in 75f055a.